tests/ui/explicit_into_iter_fn_arg.rs

```diff
@@ -0,0 +1,24 @@
+//@run-rustfix
+
+#![allow(unused)]
+#![warn(clippy::explicit_into_iter_fn_arg)]
+
+fn a<T>(_: T) {}
+fn b<T: IntoIterator<Item = i32>>(_: T) {}
+fn c(_: impl IntoIterator<Item = i32>) {}
+fn d<T>(_: T)
+where
+    T: IntoIterator<Item = i32>,
+{
+}
+fn e<T: IntoIterator<Item = i32>>(_: T) {}
+fn f(_: std::vec::IntoIter<i32>) {}
+
+fn main() {
+    a(vec![1, 2].into_iter());
+    b(vec![1, 2].into_iter());
+    c(vec![1, 2].into_iter());
+    d(vec![1, 2].into_iter());
+    e([&1, &2, &3].into_iter().cloned());
+    f(vec![1, 2].into_iter());
```

Comment:
I'd like to see an example with two chained `.into_iter()` calls.

Reply:
Seems like that triggers [`useless_conversion`](https://rust-lang.github.io/rust-clippy/master/#useless_conversion) already:
```rs
warning: useless conversion to the same type: `std::vec::IntoIter<i32>`
  --> test.rs:24:14
   |
24 |     v.extend(vec![0].into_iter().into_iter());
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider removing `.into_iter()`: `vec![0].into_iter()`
   |
```
So... maybe we shouldn't lint at all if we see `x.into_iter().into_iter()`, to avoid linting twice?

Reply:
Yeah. That makes me wonder if we should just merge this into the `useless_conversion` lint?

Reply:
Alright, makes sense, done. Sorry, took a bit longer